From 91e999da8a925f27180dff0423d2e7ca212e2590 Mon Sep 17 00:00:00 2001
From: Adrian Insaurralde Avalos <adrianinsaval@gmail.com>
Date: Sun, 3 Mar 2024 01:47:38 -0300
Subject: [PATCH 18/18] Fix 0033576:
 BRepTools_NurbsConvertModification::NewPolygon does not return false when
 creation fails

AKA https://github.com/FreeCAD/FreeCAD/issues/9760
---
 src/BRepTools/BRepTools_NurbsConvertModification.cxx | 2 +-
 1 file changed, 1 insertion(+), 1 deletion(-)

diff --git a/src/BRepTools/BRepTools_NurbsConvertModification.cxx b/src/BRepTools/BRepTools_NurbsConvertModification.cxx
index 85a30c63c6..fe4166ded8 100644
--- a/src/BRepTools/BRepTools_NurbsConvertModification.cxx
+++ b/src/BRepTools/BRepTools_NurbsConvertModification.cxx
@@ -521,7 +521,7 @@ Standard_Boolean BRepTools_NurbsConvertModification::NewPolygon(const TopoDS_Edg
         gp_Pnt aPoint = aCurve->Value(aParam);
         newParameter(aPoint, aNewCurve, aFirst, aLast, aTol, aParam);
       }
-    }
+    } else return Standard_False;
   }
   return Standard_True;
 }
-- 
2.45.0

